src/spells.cpp

```diff
@@ -283,6 +286,9 @@ bool CombatSpell::castSpell(Creature* creature)
 
 bool CombatSpell::castSpell(Creature* creature, Creature* target)
 {
+	if (!creature || !target) {
```

Comment:
This is not a proper fix. `creature` can never be `nullptr`, because it's always passed `this` in `Monster`. 
`target` can only be `nullptr` if a `Monster`'s target changes to `nullptr` when casting a spell. The proper fix would be to recheck here: https://github.com/otland/forgottenserver/blob/master/src/monster.cpp#L778 if the target changed and is still valid.

Reply:
@djarek I don't know if this is considered to be more of a user error more than it is considered a bug, but the server crashes when you pass a ``Combat`` object instead of ``Creature`` in ``combat:execute(creature, variant)``, e.g. ``combat:execute(combat, variant)``.

If you pass other values, or if ``Creature`` is ``nil`` it won't crash or anything, but will crash if you pass a ``Combat`` object.

It'll only crash if you also include ``variant`` in the function.

Reply:
Combat.execute(combat, creature, variant) does not crash

Reply:
That's not what he implied either, `combat:execute(combat, variant)` would be the equivalent to `Combat.execute(combat, combat, variant)`

Reply:
@raymondtfr it should not crash, but I don't think it's related to this PR.

Reply:
>it should not crash, but I don't think it's related to this PR.

I think I'm having déjà vu cause I think I'd already mentioned this sometime ago in another PR.

Should I open an issue for that? Thanks for the reply.

Reply:
Yeah, please open an issue if it still occurs. The lua interface should (ideally) not cause crashes.

Reply:
Now analyzing the calls from the beginning it turns out that this method is called directly from monster.cpp, so the creature will always be valid. I had not done or analyzed this before, because for some reason I thought this method could be called not just by monster.cpp.
I'll make those changes when I get back from traveling.